h3py/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+find_package(Cython MODULE REQUIRED)
+find_package(PythonExtensions MODULE REQUIRED)
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+add_cython_target(h3core C PY3)
+add_library(h3core MODULE ${h3core} ${LIB_SOURCE_FILES} ${CONFIGURED_API_HEADER})
+python_extension_module(h3core)
+set_property(TARGET h3core PROPERTY C_STANDARD 99)
+target_compile_definitions(h3core INTERFACE $<TARGET_PROPERTY:h3,COMPILE_DEFINITIONS>)
+target_include_directories(h3core INTERFACE $<TARGET_PROPERTY:h3,INCLUDE_DIRECTORIES>)
+target_link_libraries(h3core h3)
+
+add_cython_target(util C PY3)
+add_library(util MODULE ${util} ${LIB_SOURCE_FILES} ${CONFIGURED_API_HEADER})
+python_extension_module(util)
+set_property(TARGET util PROPERTY C_STANDARD 99)
+target_compile_definitions(util PUBLIC $<TARGET_PROPERTY:h3,COMPILE_DEFINITIONS>)
+target_include_directories(util PUBLIC $<TARGET_PROPERTY:h3,INCLUDE_DIRECTORIES>)
+
+install(TARGETS h3core util LIBRARY DESTINATION h3py)
+
+
+add_cython_target(geo C PY3)
```

Comment:
Each of `geo` and `util` are the different `pyx` files? It might be good to clarify what the dependency of this block is, and extracting it to a macro to deduplicate the CMake code.

Reply:
Totally agree. I just cut and paste here. A macro would be great. Does anyone know how to do that in CMake?

Reply:
There's an example of that in the H3 CMakeLsits.txt.

It looks something like:
```
macro(my_macro_name some_arg some_other_arg)
    message("Macro called with ${some_arg} and ${some_other_arg}")
endmacro()
```

You can read more at https://cmake.org/cmake/help/latest/command/macro.html

Reply:
I macro'd this thing up, and it seems to work... but I have no idea what I'm doing. It'd be great to have someone with more comfort with Cython/CMake to sign off on this eventually. I don't know that we need to do that in this PR, but definitely before we merge the `cython` branch to `master.`